### PR TITLE
A working CMC chart

### PR DESCRIPTION
**Pull request by markry-kainos**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/ROC-5274

### Change description ###

First draft of a working CMC product chart (without legal frontend).

Tested locally in K8S.

To be used with: https://github.com/hmcts/cnp-flux-config/pull/185

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: cmc/values.yaml [hunk 1]
```diff
@@ -0,0 +1,236 @@
+tags:
+  cmc-claim-store-postgresql-pod: false # claim-store chart
+  cmc-citizen-frontend-pod: false       # claim-store chart
+  cmc-ccd-stack: true                   # claim-store chart
+  idam-pr: false                        # manually configure whitelists in IDAM until IDAM here
```
Comment (timja) [inline cmc/values.yaml]: why not use this?

File: cmc/values.yaml [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+      CMC_DB_PASSWORD: hmcts
+  ccd:
+    readinessDelay: 180
+    livenessDelay: 180
+    readinessPeriod: 60
+    livenessPeriod: 60
+    adminWeb:
+      enabled: true
+      environment:
+        NODE_TLS_REJECT_UNAUTHORIZED: 0
```
Comment (timja) [inline cmc/values.yaml]: why is this needed?

File: cmc/values.yaml [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+    caseManagementWeb:
+      enabled: true
+      environment:
+        NODE_TLS_REJECT_UNAUTHORIZED: 0
```
Comment (timja) [inline cmc/values.yaml]: and this?

File: k8s/sealed-secrets/ccd-aat-ccd-admin-web-oauth2-client-secret.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+apiVersion: bitnami.com/v1alpha1
+kind: SealedSecret
+metadata:
+  creationTimestamp: null
+  name: ccd-aat-ccd-admin-web-oauth2-client-secret
```
Comment (timja) [inline k8s/sealed-secrets/ccd-aat-ccd-admin-web-oauth2-client-secret.yaml]: I don't think you want these in the chart, default it to 123456 and then override it in the flux config for a repo

File: cmc/values.yaml (hunk 3 above)
Comment (markry-kainos) [reply cmc/values.yaml]: no cert and communicate on https to api-gateway - it didnt work without, maybe changing protocol but im sure i looked at this (quickly)

File: k8s/sealed-secrets/ccd-aat-ccd-admin-web-oauth2-client-secret.yaml (hunk 4 above)
Comment (markry-kainos) [reply k8s/sealed-secrets/ccd-aat-ccd-admin-web-oauth2-client-secret.yaml]: yep agreed

File: cmc/values.yaml (hunk 1 above)
Comment (markry-kainos) [reply cmc/values.yaml]: good point - i will use

File: cmc/values.yaml [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+          - CMC
+        users:
+          - civilmoneyclaims+ccd@gmail.com|CMC|MoneyClaimCase|open
+        userProfileDatabasePort: 5432
+        userProfileDatabaseUser: hmcts
+        userProfileDatabasePassword: hmcts
+        userProfileDatabaseName: user-profile
+      definition:
+        enabled: true
+        image: hmcts.azurecr.io/hmcts/cmc-ccd-definition-importer:1.2.6.AKS
```
Comment (timja) [inline cmc/values.yaml]: is this not in hmctspublic?

File: cmc/values.yaml [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+    environment:
+      FILE_CLEANUP_ENABLED: true
+      LETTER_TRACKING_DB_CONN_OPTIONS: ""
+      LETTER_TRACKING_DB_NAME: send_letter
+      LETTER_TRACKING_DB_USER_NAME: "hmcts"
+      LETTER_TRACKING_DB_PASSWORD: "hmcts"
+
+dm-store:
+  java:
+    image: hmcts.azurecr.io/hmcts/dm-store:pr-334  # fix for postgres bug
```
Comment (timja) [inline cmc/values.yaml]: is this still only on PR?

File: k8s/bin/vault-to-sealedsecret.sh [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    'cmc|system-update-password'
+    'cmc|ccd-importer-username' 
+    'cmc|ccd-importer-password' 
+    'ccd|ccd-api-gateway-oauth2-client-secret'
+    'ccd|ccd-admin-web-oauth2-client-secret'
+    'ccpay|gov-pay-keys-cmc'
+)
+
+for i in "${secrets[@]}"; do
+    vault=`echo "$i" | awk -F'|' '{print $1}'`-$env
```
Comment (timja) [inline k8s/bin/vault-to-sealedsecret.sh]: normally you avoid backticks as they have sub weird limitations, sub-shell is safer to use
```suggestion
    vault=$(echo "$i" | awk -F'|' '{print $1}')-$env
```

File: k8s/bin/vault-to-sealedsecret.sh [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    name=`echo "$i" | awk -F'|' '{print $2}'`
```
Comment (timja) [inline k8s/bin/vault-to-sealedsecret.sh]:
```suggestion
    name=$(echo "$i" | awk -F'|' '{print $2}')
```

File: k8s/bin/vault-to-sealedsecret.sh [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    secret=`az keyvault secret show --vault-name $vault --name $name -o tsv --query value`
```
Comment (timja) [inline k8s/bin/vault-to-sealedsecret.sh]:
```suggestion
    secret=$(az keyvault secret show --vault-name $vault --name $name -o tsv --query value)
```

File: cmc/values.yaml (hunk 5 above)
Comment (markry-kainos) [reply cmc/values.yaml]: nope - im adding a ticket to check - i think there is a handful of repos that can be moved tbh

File: cmc/values.yaml (hunk 6 above)
Comment (markry-kainos) [reply cmc/values.yaml]: yessssssss 🙈 i will chase
